pkgs/by-name/ko/koboldcpp/package.nix

```
@@ -28,38 +29,53 @@
 
   metalSupport ? stdenv.isDarwin && stdenv.isAarch64,
 
+  # You can find a full list here: https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
+  # For example if you're on an GTX 1080 that means you're using "Pascal" and you need to pass "6.0"
+  cudaCapabilities ? cudaPackages.cudaFlags.cudaCapabilities or [ ],
```

`or [ ]` is unnecessary, we guarantee this attribute to exist; that said seems like what you end up using is more like

```suggestion
cudaCapabilities ? lib.take 1 cudaPackages.flags.realArches,
```

E.g.

```
❯ nix repl -f '<nixpkgs>'
...
nix-repl> lib.take 1 cudaPackages.flags.realArches
[ "sm_60" ]
```

However, are you sure you want to use `sm_` and not `compute_`?

---

Looking at: https://github.com/LostRuins/koboldcpp/blob/7499a6bd4b714aa1bb43284ed87ccf072faa3825/examples/llava/MobileVLM-README.md?plain=1#L197

It it seems like that's the way to do it? I will go with `builtins.head cudaPackages.cudaFlags.realArches`

---

Sure, but to be clear, this will be (unless upstream sneaks in some more flags in the buid script) only compatible with a single generation of GPUs, i.e. I think this will leave out the PTX code which could be jit-ed at runtime for other/newer GPUs

---

I'm a bit unsure of what I can do here. We can't grab the actual `arch` the host has, so this only really leaves us with the choice to grab the lowest one...